test-suite/shared/src/test/scala/org/scalajs/testsuite/compiler/NullPointersTest.scala

```diff
@@ -0,0 +1,289 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.compiler
+
+import org.junit.Test
+import org.junit.Assert._
+import org.junit.Assume._
+
+import org.scalajs.testsuite.utils.AssertThrows.assertThrows
+import org.scalajs.testsuite.utils.Platform._
+
+class NullPointersTest {
+  import NullPointersTest._
+
+  // Instantiate Tester somewhere, otherwise plenty of tests are moot
+  new Tester(0)
+
+  @noinline
+  private def nullOf[T >: Null]: T = null
+
+  @inline
+  private def inlineNullOf[T >: Null]: T = null
+
+  @noinline
+  private def assertNPE[U](body: => U): Unit =
+    assertThrows(classOf[NullPointerException], body)
+
+  private def assumeCompliantNPEs(): Unit =
+    assumeTrue("assuming compliant null pointer checks", hasCompliantNullPointers)
```

Comment:
Annotate this with `@BeforeClass` to make it clear that the entire test assumes NPEs (you can also then remove it from the individual tests).